src/Symfony/Component/Routing/UrlHelper.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Routing;
```

Comment:
It is weird to have this class in Routing, when it required dependency is from HttpFoundation and an optional one on Routing.

Reply:
btw, this is a BC break for the HttpFoundationExtension, as it becomes unusable without the routing component.